tools/pre-commit

```diff
@@ -11,67 +12,39 @@
 # are correctly clang-formatted and that CMakeLists.txt and *.cmake
 # files are cmake-formatted
 
-# To use this hook, you must have clang-format and cmake-format
-# installed on your system
-
 # To install this hook, symlink this hook to your git hooks as follows
 # (note that the extra ../../ in the path is because git runs the hook
 # from the .git/hooks directory, so the symlink has to be redirected)
 # ln -s -f ../../tools/pre-commit .git/hooks/pre-commit
 
-# @todo : add support for the pika inspect tool to be run as well
-
-CLANG_FORMAT_VERSION=clang-format-16
-
-red=$(tput setaf 1)
-green=$(tput setaf 2)
-yellow=$(tput setaf 3)
-blue=$(tput setaf 4)
-normal=$(tput sgr0)
-
-cxxfiles=()
-for file in `git diff --cached --name-only --diff-filter=ACMRT | grep -E "\.(cpp|hpp)$"`; do
-    if ! cmp -s <(git show :${file}) <(git show :${file}|$CLANG_FORMAT_VERSION -style=file); then
-        cxxfiles+=("${file}")
-    fi
-done
+# To use this hook, you must have podman installed.
 
-cmakefiles=()
-for file in `git diff --cached --name-only --diff-filter=ACMRT | grep -E "(CMakeLists\.txt|\.cmake)$"`; do
-    tmpfile=$(mktemp /tmp/cmake-check.XXXXXX)
-    git show :${file} > $tmpfile
-    cmake-format -c $(pwd)/.cmake-format.py -i $tmpfile
-    if ! cmp -s <(git show :${file}) <(cat $tmpfile); then
-        cmakefiles+=("${file}")
-    fi
-    rm $tmpfile
-done
-
-returncode=0
-full_list=
-
-if [ -n "${cxxfiles}" ]; then
-    printf "# ${blue}clang-format ${red}error pre-commit${normal} : To fix run the following (use git commit ${yellow}--no-verify${normal} to bypass)\n"
-    for f in "${cxxfiles[@]}" ; do
-        rel=$(realpath --relative-to "./$GIT_PREFIX" $f)
-        printf "$CLANG_FORMAT_VERSION -i %s\n" "$rel"
-        full_list="${rel} ${full_list}"
-    done
-    returncode=1
-fi
+# @todo : add support for the pika inspect tool to be run as well
 
-if [ -n "${cmakefiles}" ]; then
-    printf "# ${green}cmake-format ${red}error pre-commit${normal} : To fix run the following (use git commit ${yellow}--no-verify${normal} to bypass)\n"
-    for f in "${cmakefiles[@]}" ; do
-        rel=$(realpath --relative-to "./$GIT_PREFIX" $f)
-        printf "cmake-format -i %s\n" "$rel"
-        full_list="${rel} ${full_list}"
-    done
-    returncode=1
+function error_msg() {
+    echo "$1 formatted files. run git add -u and git commit again."
+    echo "run git commit --no-verify to bypass."
+    git status
+    exit_code=1
+}
+
+echo "Run clang-format"
+podman run -w /pika/source -v=$(git rev-parse --show-toplevel):/pika/source docker.io/pikaorg/pika-ci-base:latest \
```

Comment:
Possibly:
```suggestion
podman run -w /pika/source -v=$(git rev-parse --show-toplevel):/pika/source docker.io/pikaorg/pika-ci-base:27 \
```

It'll be right 99% of the time with `:latest` as well, but we have to update all the other `:27`s to `:28` anyway when bumping the image tag, so not the end of the world to update it here as well? On the other hand, CI will anyway reject the PR in the 1% cases. I'm fine either way.

I would maybe move the image name out into a variable though, just to make it more visible which image we're using and not having to repeat it.